src/gluonnlp/metric/length_normalized_loss.py

```diff
@@ -0,0 +1,66 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+""" Length Normalized Loss """
+
+from mxnet import ndarray
+from .masked_accuracy import EvalMetric
+
+__all__ = ['LengthNormalizedLoss']
+
+class LengthNormalizedLoss(EvalMetric):
```

Comment:
Why use evaluation metric base class for a loss?

Reply:
`EvalMetric` is the metric base class. Every training/validation metric needs to inherit from the class, see https://github.com/apache/incubator-mxnet/blob/3e638b4fc4de959c7afc614facc336a796347c9d/python/mxnet/metric.py#L1661 Although it has `Eval` as the prefix, It does not directly related to validation. I have to admit the name is confusing.

BTW `MaskedAccuracy` also inherited from `EvalMetric` although it can only be used during training. See https://github.com/dmlc/gluon-nlp/blob/434187eb5e7bb898482021ffbc9c1648d0700608/src/gluonnlp/metric/masked_accuracy.py.

We may submit an issue to rename the class `EvalMetric`.

Reply:
But here you are using some gluon-nlp EvalMetric and not the mxnet one `https://github.com/dmlc/gluon-nlp/blob/2db35f7e909e27d91b85abb8b959416688080296/src/gluonnlp/metric/masked_accuracy.py#L24`?

Reply:
the code of gluon-nlp `EvalMetric` is exactly the same as the mxnet one. I do not know why they copy and paste the `EvalMetric` code here. Because `MaskedAccuracy` inherit from this `EvalMetric`, I inherit my class from this one for consistency.

Reply:
Thanks for clarification. I looked at the PR that introduced the copy: https://github.com/dmlc/gluon-nlp/pull/503#discussion_r247326960

It's safe to remove the copy of `EvalMetric` in GluonNLP now and to replace it with the import from MXNet.

Reply:
OK, I will remove that